src/lib/patterns.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const glob = require('fast-glob');
+const fs = require('fs');
+const path = require('path');
+const matter = require('gray-matter');
+const md = require('markdown-it')();
+
+const stripDot = /^\./;
+const basePath = 'src/site/content/en/patterns';
```

Comment:
I'd probably do something like:

```js
const basePath = path.join(__dirname, '../../src/site/content/en/patterns');
```

... in case this is run from somewhere not the top-level folder.

Reply:
I'm also curious about plans for non-en, but I'm also happy to hear "none" for now, in that we have no plans to translate API docs either.

Reply:
None for now.